### PR TITLE
Fix NPEs due to missing connect.index and connect.type for sink only …

### PR DESCRIPTION
**Pull request by mohitpali**
…cases

*Issue #, if available:* 135

*Description of changes:* Avoiding Null Pointer Exception when looking for index

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java [hunk 1]
```diff
@@ -50,7 +50,12 @@ public TypeConverter get(@NonNull org.everit.json.schema.Schema jsonSchema,
         if (jsonSchema instanceof BooleanSchema) {
             typeConverter = get(Schema.Type.BOOLEAN);
         } else if (jsonSchema instanceof NumberSchema) {
-            typeConverter = get(Schema.Type.valueOf(connectType.toUpperCase()));
+            // If no connect type passed then assume that connect schema is for INT64 type data
+            if (connectType == null) {
+                typeConverter = get(Schema.Type.valueOf("INT64"));
```
Comment (blacktooth) [inline jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java:55]: Why is it so?

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/StructTypeConverter.java [hunk 2]
```diff
@@ -336,6 +336,21 @@ public SchemaBuilder toConnectSchema(org.everit.json.schema.Schema jsonSchema,
...
+            // Producer -> Kafka Topic -> Sink Connector -> Destination
+            //       |                     ^
+            //       v                     |
+            //       AWS Glue Schema Registry
+
+            if (index1 == null && index2 == null) {
+                return 0;
+            }
+            if (index1 == null) return 1;
+            if (index2 == null) return -1;
```
Comment (blacktooth) [inline jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/StructTypeConverter.java:352]: Have you validated if there are other places where we expect these properties to be present?

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java (hunk 1 above)
Comment (mohitpali) [reply jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java:55]: If no connect type passed then assume that connect schema is for INT64 type data. This is for cases where schema is not enriched schema created by GSR converters.

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/StructTypeConverter.java (hunk 2 above)
Comment (mohitpali) [reply jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/StructTypeConverter.java:352]: yes, it's only used for struct types

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java (hunk 1 above)
Comment (blacktooth) [reply jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java:55]: What's the reasoning behind INT64?

File: jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java [hunk 3]
```diff
@@ -50,7 +50,12 @@ public TypeConverter get(@NonNull org.everit.json.schema.Schema jsonSchema,
         if (jsonSchema instanceof BooleanSchema) {
             typeConverter = get(Schema.Type.BOOLEAN);
         } else if (jsonSchema instanceof NumberSchema) {
-            typeConverter = get(Schema.Type.valueOf(connectType.toUpperCase()));
+            // If no connect type passed then assume that connect schema is for FLOAT64 type data
+            if (connectType == null) {
+                typeConverter = get(Schema.Type.valueOf("FLOAT64"));
```
Comment (blacktooth) [inline jsonschema-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/jsonschema/typeconverters/TypeConverterFactory.java:55]: Minor: Can we add metadata similar to https://github.com/awslabs/aws-glue-schema-registry/blob/kafka-connect-converter-protobuf/protobuf-kafkaconnect-converter/README.md ?

**Review by blacktooth — APPROVED** [review]

**Pull request merged**